src/sys/termios.rs

```diff
@@ -187,39 +191,19 @@ impl Termios {
     ///
     /// This is not part of `nix`'s public API because it requires additional work to maintain type
     /// safety.
-    pub(crate) fn get_libc_termios(&self) -> Ref<libc::termios> {
-        {
-            let mut termios = self.inner.borrow_mut();
-            termios.c_iflag = self.input_flags.bits();
-            termios.c_oflag = self.output_flags.bits();
-            termios.c_cflag = self.control_flags.bits();
-            termios.c_lflag = self.local_flags.bits();
-            termios.c_cc = self.control_chars;
-        }
-        self.inner.borrow()
-    }
-
-    /// Exposes the inner `libc::termios` datastore within `Termios`.
-    ///
-    /// This is unsafe because if this is used to modify the inner `libc::termios` struct, it will
-    /// not automatically update the safe wrapper type around it. In this case it should also be
-    /// paired with a call to `update_wrapper()` so that the wrapper-type and internal
-    /// representation stay consistent.
-    pub(crate) unsafe fn get_libc_termios_mut(&mut self) -> *mut libc::termios {
-        {
-            let mut termios = self.inner.borrow_mut();
-            termios.c_iflag = self.input_flags.bits();
-            termios.c_oflag = self.output_flags.bits();
-            termios.c_cflag = self.control_flags.bits();
-            termios.c_lflag = self.local_flags.bits();
-            termios.c_cc = self.control_chars;
-        }
-        self.inner.as_ptr()
+    pub(crate) fn get_libc_termios(&self) -> libc::termios {
+        let mut termios = self.inner.clone();
```

Comment:
You shouldn't need `clone` if libc::termios is already `Copy`.  And if `libc::termios` is not already `Copy`, then you can't derive `Copy`.  Or is it only `Copy` on some platforms?

Reply:
Nothing subtle, I just missed it. I believe it is clone on all platforms.

Reply:
Pushed a version without the clone.